02_classify_mutations/predict_all_genes.py

```diff
@@ -0,0 +1,218 @@
+    io_args, model_options = arg_groups['io'], arg_groups['model_options']
+
+    # add some additional hyperparameters/ranges from config file to model options
+    # these shouldn't be changed by the user, so they aren't added as arguments
+    model_options.n_dim = None
+    model_options.alphas = cfg.alphas
+    model_options.l1_ratios = cfg.l1_ratios
+    model_options.standardize_data_types = cfg.standardize_data_types
+    model_options.shuffle_by_cancer_type = cfg.shuffle_by_cancer_type
+
+    return io_args, model_options
+
+
+if __name__ == '__main__':
+
+    # process command line arguments
+    io_args, model_options = process_args()
+    sample_info_df = du.load_sample_info(model_options.training_data,
+                                         verbose=io_args.verbose)
+
+    # create results dir and subdir for experiment if they don't exist
+    experiment_dir = Path(io_args.results_dir, 'gene').resolve()
+    experiment_dir.mkdir(parents=True, exist_ok=True)
+
+    # save model options for this experiment
+    # (hyperparameters, preprocessing info, etc)
+    fu.save_model_options(experiment_dir, model_options)
+
+    # create empty log file if it doesn't exist
+    log_columns = [
+        'gene',
+        'training_data',
+        'skip_reason'
+    ]
+    if io_args.log_file.exists() and io_args.log_file.is_file():
+        log_df = pd.read_csv(io_args.log_file, sep='\t')
+    else:
+        log_df = pd.DataFrame(columns=log_columns)
+        log_df.to_csv(io_args.log_file, sep='\t')
+
+    tcga_data = TCGADataModel(seed=model_options.seed,
+                              subset_mad_genes=model_options.subset_mad_genes,
+                              training_data=model_options.training_data,
+                              overlap_data_types=model_options.overlap_data_types,
+                              load_compressed_data=True,
+                              standardize_input=True,
+                              n_dim=N_DIM,
+                              sample_info_df=sample_info_df,
+                              verbose=io_args.verbose)
+    genes_df = tcga_data.load_gene_set('vogelstein')
+
+    progress = tqdm(genes_df.iterrows(),
+                    total=genes_df.shape[0],
+                    ncols=100,
+                    file=sys.stdout)
+
+    all_genes = []
+    all_preds = []
+    sample_list = None
+
+    for gene_idx, gene_series in progress:
+        log_df = None
+        gene = gene_series.gene
+        classification = gene_series.classification
+        progress.set_description('gene: {}'.format(gene))
+
+        try:
+            tcga_data.process_data_for_gene(gene,
+                                            classification,
+                                            experiment_dir,
+                                            filter_cancer_types=False)
+        except KeyError:
+            # this might happen if the given gene isn't in the mutation data
+            # (or has a different alias)
```

Comment:
In what cases do the genes have different aliases? Should there be a step for gene id conversion?